### DISCUSSION TITLE
File extension

### DISCUSSION THREAD

**Discussion by atikjaz** [spencerwooo/onedrive-vercel-index#663] [Ideas]
Can we hide the `.md` extension in title, it will look like `README` only and it will also preview

**Comment by myl7** [comment]
The README reuses [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) to render, which is just like other Markdown files, so hidding `.md` seems not to make sense for this upstream. You can edit [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) and do the check to implement it on your fork though.

**Reply by adhiduzz** [reply]
any idea how to hide .password file?

**Reply by myl7** [reply]
See https://github.com/spencerwooo/onedrive-vercel-index/discussions/120#discussioncomment-1375027

Though the text is not in English, the code should be explicit.

**Reply by adhiduzz** [reply]
where should i put that code?

edited: i know it's written on filelisting.tsx
what line should i put that code?
at the end of line 231 or replace or something else?

**Reply by atikjaz** [reply]
> The README reuses [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) to render, which is just like other Markdown files, so hidding `.md` seems not to make sense for this upstream. You can edit [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) and do the check to implement it on your fork though.

Okay.
Is there any way we can hide download button as it will look like reading book no copy link download button etc.

**Reply by myl7** [reply]
> where should i put that code?
> 
> edited: i know it's written on filelisting.tsx what line should i put that code? at the end of line 231 or replace or something else?

Oh there are already some code changes since then. The ref is just for reference.
As for now you can edit:

https://github.com/spencerwooo/onedrive-vercel-index/blob/152380bd057d256bcd6cf3ececefcf5826fecaf9/components/FolderListLayout.tsx#L96

https://github.com/spencerwooo/onedrive-vercel-index/blob/152380bd057d256bcd6cf3ececefcf5826fecaf9/components/FolderGridLayout.tsx#L108

to filter out `.password` according to `c.id`. It does require some fundamental React knowledge to do it corretly though.

**Reply by myl7** [reply]
> > The README reuses [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) to render, which is just like other Markdown files, so hidding `.md` seems not to make sense for this upstream. You can edit [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) and do the check to implement it on your fork though.
> 
> Okay. Is there any way we can hide download button as it will look like reading book no copy link download button etc.

You can iterate over all preview components in [`components/previews`](https://github.com/spencerwooo/onedrive-vercel-index/tree/main/components/previews) to hide via addding class `hidden` / remove `DownloadBtnGtoup` component.

**Reply by atikjaz** [reply]
> > > The README reuses [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) to render, which is just like other Markdown files, so hidding `.md` seems not to make sense for this upstream. You can edit [`MarkdownPreview`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/components/previews/MarkdownPreview.tsx) and do the check to implement it on your fork though.
> > 
> > 
> > Okay. Is there any way we can hide download button as it will look like reading book no copy link download button etc.
> 
> You can iterate over all preview components in [`components/previews`](https://github.com/spencerwooo/onedrive-vercel-index/tree/main/components/previews) to hide via addding class `hidden` / remove `DownloadBtnGtoup` component.

Where to add hidden? I'm not good at coding. Line number?

**Reply by adhiduzz** [reply]
@myl7 so, it doesn't implemented for the latest build?

**Reply by myl7** [reply]
> it doesn't implemented for the latest build

Yes it is not implemented on this upstream.
If you really need to hide `.password`, you can use [my fork](https://github.com/myl7/onedrive-vercel-index) which adds an option to enable it.

**Reply by myl7** [reply]
> Where to add hidden? I'm not good at coding. Line number?

Well, it actually does require some React experience to implement it. And since your requirement (hide some buttons) is hard to specified into config options, I might be unable to help much.
